File: src/main/java/com/sleepkqq/sololeveling/view/auth/TgWebAppData.java
package com.sleepkqq.sololeveling.view.auth;

import com.fasterxml.jackson.databind.PropertyNamingStrategies;
import com.fasterxml.jackson.databind.annotation.JsonNaming;
import java.time.LocalDateTime;

@JsonNaming(PropertyNamingStrategies.SnakeCaseStrategy.class)
public record TgWebAppData(
    LocalDateTime authDate,
    String chatType,
    String chatInstance,
    String hash,
    String signature,
    TgUser user
) {

}

File: soloist-gateway-service/src/main/kotlin/com/soloist/gateway/dto/auth/TgWebAppData.kt
package com.soloist.gateway.dto.auth

import tools.jackson.databind.PropertyNamingStrategies
import tools.jackson.databind.annotation.JsonNaming
import java.time.OffsetDateTime

@JsonNaming(PropertyNamingStrategies.SnakeCaseStrategy::class)
data class TgWebAppData(
	val queryId: String?,
	val user: TgUserData,
	val authDate: OffsetDateTime,
	val hash: String,
	val signature: String?,
	val startParam: String?
)
